packages/state/src/lib/transactions.ts

```diff
@@ -300,3 +301,17 @@ export function transact<T>(fn: () => T): T {
 	}
 	return transaction(fn)
 }
+
+/**
+ * @internal
+ */
+export function pauseReactions() {
+	assert(!inst.currentTransaction, 'Cannot pause reactions inside a transaction')
+	const txn = (inst.currentTransaction = new Transaction(null))
+
+	return () => {
+		assert(txn === inst.currentTransaction, 'Cannot resume reactions while in a transaction')
+		inst.currentTransaction = null
+		txn.commit()
+	}
+}
```

Comment:
This was allows us to prevent rerendering the app in the middle of a mutator, which might lead to inconsistent/buggy states.

Reply:
good catch! i'm gonna remove this for now and add a follow up ticket to do it properly later. The place where I was seeing issues that required this (adding a new file not triggering the input to be focused) was really caused by something else (bad immutable data handling causing too many rerenders and then us relying on that happening) which I solved separately. So it should be fine to add this later.